From 2868db5dc838cf78145d5d8f18c31505f068fc00 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Mon, 2 Sep 2024 14:30:13 +1100
Subject: [PATCH 3/3] Revert "Declare _availability_version_check as
 weak_import instead of looking it"

This reverts commit b653a2823fe4b4c9c6d85cfe119f31d8e70c2fa0.
---
 compiler-rt/lib/builtins/os_version_check.c | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/compiler-rt/lib/builtins/os_version_check.c b/compiler-rt/lib/builtins/os_version_check.c
index 182eabe7a6ae..ebfb2dfc72dd 100644
--- a/compiler-rt/lib/builtins/os_version_check.c
+++ b/compiler-rt/lib/builtins/os_version_check.c
@@ -86,10 +86,6 @@ typedef Boolean (*CFStringGetCStringFuncTy)(CFStringRef, char *, CFIndex,
                                             CFStringEncoding);
 typedef void (*CFReleaseFuncTy)(CFTypeRef);
 
-extern __attribute__((weak_import))
-bool _availability_version_check(uint32_t count,
-                                 dyld_build_version_t versions[]);
-
 static void _initializeAvailabilityCheck(bool LoadPlist) {
   if (AvailabilityVersionCheck && !LoadPlist) {
     // New API is supported and we're not being asked to load the plist,
@@ -98,8 +94,8 @@ static void _initializeAvailabilityCheck(bool LoadPlist) {
   }
 
   // Use the new API if it's is available.
-  if (_availability_version_check)
-    AvailabilityVersionCheck = &_availability_version_check;
+  AvailabilityVersionCheck = (AvailabilityVersionCheckFuncTy)dlsym(
+      RTLD_DEFAULT, "_availability_version_check");
 
   if (AvailabilityVersionCheck && !LoadPlist) {
     // New API is supported and we're not being asked to load the plist,
